### PR TITLE
8340946: Add vmTestbase/gc/memory/Nio/Nio.java and java/nio/Buffer/LimitDirectMemory.java to problem list

### PR DESCRIPTION
**Pull request by bplb**
Put tests on problem list until they are fixed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340946](https://bugs.openjdk.org/browse/JDK-8340946): Add vmTestbase/gc/memory/Nio/Nio.java and java/nio/Buffer/LimitDirectMemory.java to problem list (**Sub-task** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**) Review applies to [fec5e2ff](https://git.openjdk.org/jdk/pull/21186/files/fec5e2ffb3316855664afe8e20140635acdc2361)
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21186/head:pull/21186` \
`$ git checkout pull/21186`

Update a local copy of the PR: \
`$ git checkout pull/21186` \
`$ git pull https://git.openjdk.org/jdk.git pull/21186/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21186`

View PR using the GUI difftool: \
`$ git pr show -t 21186`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21186.diff">https://git.openjdk.org/jdk/pull/21186.diff</a>

</details>

### CODE REVIEW

**Review by AlanBateman — APPROVED** [review]

File: test/hotspot/jtreg/ProblemList.txt [hunk 1]
```diff
@@ -174,6 +174,7 @@ vmTestbase/nsk/jvmti/scenarios/capability/CM03/cm03t001/TestDescription.java 807
 vmTestbase/nsk/jvmti/InterruptThread/intrpthrd003/TestDescription.java 8288911 macosx-all
 
 vmTestbase/gc/lock/jni/jnilock002/TestDescription.java 8192647 generic-all
+vmTestbase/gc/memory/Nio/Nio.java 8340946 generic-all
```
Comment (liach) [inline test/hotspot/jtreg/ProblemList.txt:177]:
```suggestion
vmTestbase/gc/memory/Nio/Nio.java 8340728 generic-all
```

File: test/jdk/ProblemList.txt [hunk 2]
```diff
@@ -563,6 +563,8 @@ java/net/Socket/asyncClose/Race.java                            8317801 aix-ppc6
 
 # jdk_nio
 
+java/nio/Buffer/LimitDirectMemory.java                          8340946 generic-all
```
Comment (liach) [inline test/jdk/ProblemList.txt:566]:
```suggestion
java/nio/Buffer/LimitDirectMemory.java                          8340728 generic-all
```

**Review by liach — CHANGES_REQUESTED** [review]

**Review by liach — APPROVED** [review]
Bug ID looks right.

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up. This is a trivial fix.

Although there are no macosx-x64 sightings, I think using generic-all
is the right choice.

**Comment by bplb** [comment]
> Although there are no macosx-x64 sightings, I think using generic-all
> is the right choice.

I have reproduced both failures locally on my Mac.

**Pull request closed without merging**